pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -200,59 +218,147 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	// TODO: Implement also logic for generate VM name
 	// Check if this VM already exists
 	found := &kubevirtv1.VirtualMachine{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: vmSpec.Name, Namespace: vmSpec.Namespace}, found)
 	if err != nil && errors.IsNotFound(err) {
+		// Update condition to create VM:
+		cond := conditions.NewProccessingCondition(string(v2vv1alpha1.CreatingTargetVM), "Creating virtual machine")
+		err = r.upsertStatusCondition(request.NamespacedName, cond)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 
 		// Create kubevirt VM from source VM:
 		reqLogger.Info("Creating a new VM", "VM.Namespace", vmSpec.Namespace, "VM.Name", vmSpec.Name)
 		err = r.client.Create(context.TODO(), vmSpec)
 		if err != nil {
+			// Update condition
+			cond = conditions.NewSucceededCondition(string(v2vv1alpha1.VMCreationFailed), fmt.Sprintf("Error while creating virtual machine: %s", err))
+			err = r.upsertStatusCondition(request.NamespacedName, cond)
+			// Cleanup
+			if err = r.afterFailure(request.NamespacedName, provider); err != nil {
+				return reconcile.Result{}, err
+			}
 			return reconcile.Result{}, err
 		}
 
-		// Secret with username/password for the image import:
-		dvCreds := provider.GetDataVolumeCredentials()
-		if err = r.ensureDVSecretExists(instance, vmSpec.Namespace, dvCreds); err != nil {
+		// Set target name:
+		if err = r.updateTargetVMName(request.NamespacedName, vmSpec.Name); err != nil {
 			return reconcile.Result{}, err
 		}
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
 
-		// CM containing CA for the image import:
-		if err = r.ensureDVConfigMapExists(instance, vmSpec.Namespace, dvCreds); err != nil {
-			return reconcile.Result{}, err
-		}
+	// Import disks:
+	dvs := mapper.MapDisks()
+	dvsDone := make(map[string]bool)
+	for dvID := range dvs {
+		foundDv, err := r.kubeClient.CdiClient().CdiV1alpha1().DataVolumes(instance.Namespace).Get(dvID, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			// Create it:
+			// Secret with username/password for the image import:
+			dvCreds := provider.GetDataVolumeCredentials()
+			if err = r.ensureDVSecretExists(instance, vmSpec.Namespace, dvCreds); err != nil {
```

Comment:
Why in the loop? If there are multiple dvs, that method will be called for each of them in the first reconciliation (and that entails multiple secret retrievals). 
Of course, it will be called zero times if the dvs are already being imported vs. always once outside of the loop.:)

Reply:
Well, right. I think better place will be to call it, in VM creation block, because it's called only once.